atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/table/WideRowTableWithAbortedValues.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.performance.benchmarks.table;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.ptobject.EncodingUtils;
+import com.palantir.util.crypto.Sha256Hash;
+
+/**
+ * State class for creating a single Atlas table with one wide row.
+ * Unlike {@link WideRowTable}, this table may have columns which were never successfully committed, as well as
+ * columns which have committed versions, but have newer uncommitted versions on top.
+ */
+public abstract class WideRowTableWithAbortedValues extends WideRowTable {
+    public static final byte[] DUMMY_VALUE = PtBytes.toBytes("dummy");
+
+    public abstract int getNumColsCommitted();
+    public abstract int getNumColsCommittedAndNewerUncommitted();
+    public abstract int getNumColsUncommitted();
+
+    public abstract int getNumUncommittedValues();
```

Comment:
I'd add a bit more explanation that we write into the cells that have uncommitted writes this many times

Reply:
I've added some javadoc that should hopefully make things a lot clearer, and renamed this to `getNumUncommittedValuesPerCell` as well.